Artifacts/windows-visualstudio/artifact.ps1

```diff
@@ -43,78 +90,133 @@ function DownloadToFilePath ($downloadUrl, $targetFile)
+            $downloadUrl = 'https://download.microsoft.com/download/F/3/4/F3478590-7B38-48B1-BB6E-3141A9A155E7/vs_Enterprise.exe'
+        }
     }
-    elseif($sku -eq 'Enterprise') {
-        $argumentList = "--quiet --norestart --wait --add Microsoft.VisualStudio.Workload.Azure --add Microsoft.VisualStudio.Workload.Data --add Microsoft.VisualStudio.Workload.ManagedDesktop --add Microsoft.VisualStudio.Workload.NativeDesktop --add Microsoft.VisualStudio.Workload.NativeCrossPlat --add Microsoft.VisualStudio.Workload.NetCoreTools --add Microsoft.VisualStudio.Workload.NetWeb --add Component.GitHub.VisualStudio --add Microsoft.VisualStudio.Component.TestTools.MicrosoftTestManager --add Microsoft.VisualStudio.Component.TestTools.WebLoadTest --includeRecommended"
-        $downloadUrl = 'https://download.microsoft.com/download/F/3/4/F3478590-7B38-48B1-BB6E-3141A9A155E7/vs_Enterprise.exe'
+    else
+    {
+        throw "Version is not recognized - allowed values are 2015 and 2017. Specified value: $version"
     }
-}
-else
-{
-    Write-Error "Version is not recognized - allowed values are 2015 and 2017. Specified value: $version"
-}
 
-$localFile = Join-Path $logFolder 'vsinstaller.exe'
-DownloadToFilePath $downloadUrl $localFile
+    $localFile = Join-Path $logFolder 'vsinstaller.exe'
+    DownloadToFilePath $downloadUrl $localFile
 
-if(![String]::IsNullOrWhiteSpace($installerArgs))
-{
-    Write-Output "InstallerArgs value: $installerArgs"
-    $argumentList = "$installerArgs $argumentList"
-}
+    # Ensure there are no duplicate entries in the argument list.
+    $argumentList = $argumentList | Select -Unique
 
-Write-Output "Running install with the following arguments: $argumentList"
-$retCode = Start-Process -FilePath $localFile -ArgumentList $argumentList -Wait -PassThru
+    Write-Output "Running install with the following arguments: $argumentList"
+    $retCode = Start-Process -FilePath $localFile -ArgumentList $argumentList -Wait -PassThru
 
-if ($retCode.ExitCode -ne 0 -and $retCode.ExitCode -ne 3010)
-{
-    if($version -eq '2017')
+    if ($retCode.ExitCode -ne 0 -and $retCode.ExitCode -ne 3010)
     {
-        $targetLogs = 'c:\VS2017Logs'
-        New-Item -ItemType Directory -Force -Path $targetLogs | Out-Null
-        Write-Output ('Temp location is ' + $env:TEMP)
-        Copy-Item -path $env:TEMP\dd* -Destination $targetLogs
+        if($version -eq '2017')
+        {
+            $targetLogs = 'c:\VS2017Logs'
+            New-Item -ItemType Directory -Force -Path $targetLogs | Out-Null
+            Write-Output ('Temp location is ' + $env:TEMP)
+            Copy-Item -path $env:TEMP\dd* -Destination $targetLogs
+        }
+
+        throw "Product installation of $localFile failed with exit code: $($retCode.ExitCode.ToString())"    
     }
 
-    Write-Error "Product installation of $localFile failed with exit code: $($retCode.ExitCode.ToString())"    
-}
-else
-{
     Write-Output "Visual Studio install succeeded. Rebooting..."
```

Comment:
Earlier, you add the option -norestart. So this would not be rebooting, right?

Reply:
The logic is meant to allow control of the rebooting by the artifact and not the VS installer. So, in the artifact definition, we do have a restart post deploy action.